packages/nodes-base/nodes/MySql/v1/GenericFunctions.ts

```diff
@@ -34,16 +34,14 @@ export async function searchTables(
 ): Promise<INodeListSearchResult> {
 	const credentials = await this.getCredentials('mySql');
 	const connection = await createConnection(credentials);
-	const sql = `
-	SELECT table_name FROM information_schema.tables
-	WHERE table_schema = '${credentials.database}'
-		and table_name like '%${query || ''}%'
-	ORDER BY table_name
-	`;
-	const [rows] = await connection.query(sql);
+	const sql =
+		'SELECT table_name FROM INFORMATION_SCHEMA.TABLES WHERE table_schema = ? and table_name like ? ORDER BY table_name';
+
+	const values = [credentials.database, `%${query ?? ''}%`];
+	const [rows] = await connection.query(sql, values);
 	const results = (rows as IDataObject[]).map((r) => ({
-		name: r.TABLE_NAME as string,
-		value: r.TABLE_NAME as string,
+		name: r.table_name as string,
+		value: r.table_name as string,
```

Comment:
are we sure about case? maybe better like in `listSearch`?
```
			name: (table.table_name as string) || (table.TABLE_NAME as string),
			value: (table.table_name as string) || (table.TABLE_NAME as string),
```

Reply:
I could put it back to uppercase but it is working with lowercase I pushed the change to fix that.

Reply:
Added it anyway as it looks cleaner.